File: C#/ToyLanguageC/utils/ExeStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageC.utils;

namespace ToyLanguageC.utils
{
    class ExeStack<E> : IExeStack<E>
    {
        Stack<E> exeStack;

        public ExeStack()
        {
            exeStack = new Stack<E>();
        }
        public List<E> getAll()
        {
            return exeStack.ToList();
        }

        public bool isEmpty()
        {
            return exeStack.Count()==0;
        }

        public E peek()
        {
            return exeStack.Peek();
        }

        public E pop()
        {
            return exeStack.Pop();
        }

        public void push(E elem)
        {
            exeStack.Push(elem);
        }

        public string toString()
        {
            string str = "";
            str += "ExecutionStackImpl = {";

            if (exeStack.Count() > 0)
            {
                str += "\n";
            }

            foreach (E statement in exeStack)
            {
                str += "   " + statement + "\n";
            }

            str += "}";
            return str;
        }
    }
}

File: ToyLanguageInterpretor/src/utils/ExeStack.java
package utils;
import java.util.*;
public class ExeStack<E> implements IExeStack<E>{
	private Deque<E> stack;
	
	public ExeStack() {
		stack = new ArrayDeque<E>();
	}
	public void push(E elem) {
		stack.push(elem);		
	}

	@Override
	public E pop() {
		return stack.pop();
	}

	@Override
	public boolean isEmpty() {
		return stack.isEmpty();
	}

	@Override
	public E top() {
		return stack.peek();
	}
	
	public String toString() {
		return stack.toString();
	}
	@Override
	public Iterable<E> getAll() {
		return stack;
	}
}
